packetnetworking/cli.py

```diff
@@ -68,6 +69,37 @@ def cli(
             )
         )
 
+    # arbitrary retry count
+    max_attempts = 10
+    retry_delay = 5
+    attempt = 1
+    while True:
+        try:
+            try_run(
+                metadata_file,
+                metadata_url,
+                operating_system,
+                rootfs,
+                resolvers,
+                verbose,
+                quiet,
+            )
+            break
+        except Exception as exc:
+            if attempt == max_attempts:
+                raise
+            attempt += 1
+            log.error(
+                "Caught unexpected exception ('{}'), retrying in {} seconds...".format(
+                    exc, retry_delay
+                )
+            )
+            time.sleep(retry_delay)
```

Comment:
I'd really like to see this become an exponential backoff rather than the linear backoff we're using here. Even a Fibonacci backoff would work.

Reply:
Ah right, forgot about that part...will do that momentarily.

Reply:
...alright, now with 2^n delay, with n capped at 7 so as to keep the overall duration to the just-give-up point "reasonable" (I think?).